aztec/src/main/kotlin/org/wordpress/aztec/AztecText.kt

```diff
@@ -462,29 +455,27 @@ open class AztecText : AppCompatEditText, TextWatcher, UnknownHtmlSpan.OnUnknown
         // https://android-review.googlesource.com/c/platform/frameworks/base/+/634929
         val dynamicLayoutCrashPreventer = InputFilter { source, start, end, dest, dstart, dend ->
             var temp : CharSequence? = null
-            if (!bypassCrashPreventerInputFilter
-                    && dstart == dend && dest.length > dend+1
-                    && source != Constants.NEWLINE_STRING) {
-                // dstart == dend means this is an insertion
-                // avoid handling anything if it's a newline
+            if (!bypassCrashPreventerInputFilter) {
+
                 // if there are any images right after the destination position, hack the text
-                val spans = dest.getSpans(dstart, dend+1, AztecImageSpan::class.java)
+                val spans = dest.getSpans(dend, dend+1, AztecImageSpan::class.java)
```

Comment:
I think we still need to check `dest.length > dend+1` before this line so `getSpans` doesn't try to look off the limits of the buffer

Reply:
I believe the `getSpans` method automatically adjusts the query boundaries to the constraints of the underlying `Spannable`, but I could not find documentation to confirm that. I've added a commit to address it, just in case.

I modified the "fence-post" slightly, as I believe the previous condition stopped short of the last character, since the span positions are counted by the space between characters. E.g. in the text below, if "!" was the image character, and "**Hello** world" was the selection, we would have `dstart == 0`, `dend == 11`, and `dest.length == 12`. In that case, the condition `dest.length > dend + 1` would avoid the filter, though a single character follows the selection.

![hello-world-spans](https://user-images.githubusercontent.com/8507675/57057450-26c46e80-6cec-11e9-8ba6-3e3f4b2a0c8d.png)

In my observations, I have seen that image (obj.) characters are followed by a `\n`, so this may be a moot point, but I am unsure if this is guaranteed to be true at all times that text is being modified.

Reply:
Ah, gotcha - you're right. I was assuming too much re: selection start, end and length. I understand it's ok then 👍